packages/eas-cli/src/build/ios/bundleIdentifer.ts

```diff
@@ -5,14 +5,34 @@ import chalk from 'chalk';
 import fs from 'fs-extra';
 
 import log from '../../log';
-import { getProjectConfigDescription } from '../../project/projectUtils';
+import {
+  ensureAppIdentifierIsDefinedAsync,
+  getProjectConfigDescription,
+} from '../../project/projectUtils';
 import { promptAsync } from '../../prompts';
+import { Platform } from '../types';
 
 enum BundleIdenitiferSource {
   XcodeProject,
   AppJson,
 }
 
+function isBundleIdentifierValid(bundleIdentifier: string): boolean {
+  return /^[a-zA-Z][a-zA-Z0-9\-.]+$/.test(bundleIdentifier);
+}
+
+export async function ensureBundleIdentifierIsValidAsync(projectDir: string) {
+  const bundleIdentifier = await ensureAppIdentifierIsDefinedAsync(projectDir, Platform.iOS);
+  if (!isBundleIdentifierValid(bundleIdentifier)) {
+    const configDescription = getProjectConfigDescription(projectDir);
+    log.error(
+      `Invalid format of iOS bundleId. Only alphanumeric characters, '.', '-', and '_' are allowed, and each '.' must be followed by a letter.`
```

Comment:
Please fix this message. `_` is not allowed in bundle ids.

> The bundle identifier string identifies your application to the system. This string must be a uniform type identifier (UTI) that contains only alphanumeric (A-Z,a-z,0-9), hyphen (-), and period (.) characters. The string should also be in reverse-DNS format. For example, if your company’s domain is Ajax.com and you create an application named Hello, you could assign the string com.Ajax.Hello as your application’s bundle identifier.

Reply:
copied from expo-cli, actual regexp is correct, the only message is wrong